internal/commands/user/list/command.go

```diff
@@ -13,8 +20,106 @@ var Command = cli.CommandDefinition{
 	Command:     &command{},
 }
 
-type command struct{}
+type command struct {
+	inputs      inputs
+	realmClient realm.Client
+	users       []realm.User
+}
+
+func (cmd *command) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+
+	fs.Var(&cmd.inputs.StateValue, flagState, flagStateUsage)
+	fs.BoolVar(&cmd.inputs.Pending, flagStatus, false, flagStatusUsage)
+	fs.Var(newProviderTypesValue(&cmd.inputs.ProviderTypes), flagProviderTypes, flagProviderTypesUsage)
+	fs.StringSliceVar(&cmd.inputs.Users, flagUsers, []string{}, flagUsersUsage)
+}
+
+func (cmd *command) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+func (cmd *command) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = realm.NewAuthClient(profile.RealmBaseURL(), profile.Session())
+	return nil
+}
 
 func (cmd *command) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := cli.ResolveApp(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+	users, usersErr := cmd.realmClient.FindUsers(
+		app.GroupID,
+		app.ID,
+		realm.UserFilter{
+			State:     cmd.inputs.StateValue.getUserState(),
+			Pending:   cmd.inputs.Pending,
+			Providers: cmd.inputs.ProviderTypes,
+			IDs:       cmd.inputs.Users,
+		},
+	)
+	if usersErr != nil {
+		return usersErr
+	}
+
+	cmd.users = users
+
 	return nil
 }
+
+const (
+	headerID                     = "ID"
+	headerName                   = "Name"
+	headerEnabled                = "Enabled"
+	headerType                   = "Type"
+	headerLastAuthenticationDate = "Last Authentication"
+)
+
+func (cmd *command) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.users) == 0 {
+		return ui.Print(terminal.NewTextLog("No available users to show"))
+	}
+	var usersByProvider = make(map[string][]realm.User, 0)
+	for _, user := range cmd.users {
+		for _, identity := range user.Identities {
+			usersByProvider[identity.ProviderType] = append(usersByProvider[identity.ProviderType], user)
+		}
+	}
+
+	var logs []terminal.Log
+	for provider, users := range usersByProvider {
+		sort.Slice(
+			users,
+			func(i, j int) bool {
+				return users[i].LastAuthenticationDate < users[j].LastAuthenticationDate
+			},
+		)
+
+		userTable := make([]map[string]interface{}, 0, len(users))
```

Comment:
talked offline on the approach here, should be straightforward enough to implement for both `api-key` and `userpass` (we might want to figure out a common place to define those values, maybe `cloud/realm/auth_provider.go`) as the same things (I think) are used in `user create` flags

Reply:
Sorry, I just saw this comment.  Can you clarify what you are saying here?  I think I might have misunderstood the offline conversation.